C++17 code found in ylm.cpp, need to eliminate it, we only support up to C++11

### Describe the Code Quality Issue

In file included from /home/510Group/2_abacus/abacus-mc/source/source_base/ylm.cpp:10:
/home/510Group/2_abacus/abacus-mc/source/source_base/ylmcoef.h:15:1: warning: inline variables are only available with ‘-std=c++17’ or ‘-std=gnu++17’
   15 | inline constexpr double ylmcoef[YLMCOEF_SIZE] = {
      | ^~~~~~


### Additional Context

_No response_

### Task list for Issue attackers (only for developers)

- [ ] Identify the specific code file or section with the code quality issue.
- [ ] Investigate the issue and determine the root cause.
- [ ] Research best practices and potential solutions for the identified issue.
- [ ] Refactor the code to improve code quality, following the suggested solution.
- [ ] Ensure the refactored code adheres to the project's coding standards.
- [ ] Test the refactored code to ensure it functions as expected.
- [ ] Update any relevant documentation, if necessary.
- [ ] Submit a pull request with the refactored code and a description of the changes made.

--- a/source/source_base/ylmcoef.h
+++ b/source/source_base/ylmcoef.h
@@ -12,7 +12,8 @@ constexpr int YLMCOEF_SIZE = 36;
 // GPU version: must use numeric constants (CUDA doesn't support runtime init for __device__)
 __device__ static const double ylmcoef[YLMCOEF_SIZE] = {
 #else
-inline constexpr double ylmcoef[YLMCOEF_SIZE] = {
+// Use static constexpr for C++11 compatibility (inline variables require C++17)
+static constexpr double ylmcoef[YLMCOEF_SIZE] = {
 #endif
 0.28209479177387813964,     // 1.0 / sqrt(ModuleBase::FOUR_PI)
 0.48860251190291992263,		// sqrt (3.0 / ModuleBase::FOUR_PI)

